.github/workflows/pre-submit.actions.yml

```diff
@@ -117,3 +123,117 @@ jobs:
+        with:
+          path: __BUILDER_CHECKOUT_DIR__
+
+      - name: Create artifact
+        run: |
+          echo artifact > artifact
+
+      - name: Upload generated binary
+        uses: ./__BUILDER_CHECKOUT_DIR__/.github/actions/secure-upload-artifact
+        with:
+          name: artifact
+          path: artifact
+
+      - name: Download artifact
+        uses: ./__BUILDER_CHECKOUT_DIR__/.github/actions/secure-download-artifact
+        with:
+          name: artifact
+          path: path/to/__BUILDER_CHECKOUT_DIR__/artifact
+          sha256: 5b3513f580c8397212ff2c8f459c199efc0c90e4354a5f3533adf0a3fff3a530
+
+  secure-download-artifact-builder-repo-folder:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b # tag=v3.0.2
+        with:
+          path: __BUILDER_CHECKOUT_DIR__
+
+      - name: Create artifact and folder
+        run: |
+          mkdir some-folder
+          echo artifact > artifact
+
+      - name: Upload generated binary
+        uses: ./__BUILDER_CHECKOUT_DIR__/.github/actions/secure-upload-artifact
+        with:
+          name: artifact
+          path: artifact
+
+      - name: Download artifact
+        uses: ./__BUILDER_CHECKOUT_DIR__/.github/actions/secure-download-artifact
+        with:
+          name: artifact
+          path: some-folder
+          sha256: 5b3513f580c8397212ff2c8f459c199efc0c90e4354a5f3533adf0a3fff3a530
+
+  secure-download-artifact-builder-repo-file:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b # tag=v3.0.2
+        with:
+          path: __BUILDER_CHECKOUT_DIR__
+
+      - name: Create artifact
+        run: |
+          echo artifact > artifact
+
+      - name: Upload generated binary
+        uses: ./__BUILDER_CHECKOUT_DIR__/.github/actions/secure-upload-artifact
+        with:
+          name: artifact
+          path: artifact
+
+      - name: Download artifact
+        uses: ./__BUILDER_CHECKOUT_DIR__/.github/actions/secure-download-artifact
+        with:
+          name: artifact
+          path: artifact
+          sha256: 5b3513f580c8397212ff2c8f459c199efc0c90e4354a5f3533adf0a3fff3a530
+
+  secure-download-artifacts:
+    runs-on: ubuntu-latest
+    needs: [secure-download-artifact, secure-download-artifact-builder-repo-name, secure-download-artifact-builder-repo-folder, secure-download-artifact-builder-repo-file]
+    if: ${{ always() }}
+    env:
+      BUILDER_NAME_RESULT: ${{ needs.secure-download-artifact-builder-name.result }}
+      BUILDER_FOLDER_RESULT: ${{ needs.secure-download-artifact-builder-repo-folder.result }}
+      BUILDER_FILE_RESULT: ${{ needs.secure-download-artifact-builder-repo-file.result }}
+      DOWNLOAD_RESULT: ${{ needs.secure-download.result }}
+    steps:
+      - run: |
+          # exit 0 if checks were successful.
+          [ "${BUILDER_NAME_RESULT}" != "success" ] && \
```

Comment:
nit: instead of using '&&' you could just have one of these checks on each line since they will continue to execute as long as the exit codes are zero.

Reply:
done. I added an different exit code to simply troubleshooting.